dask/blockwise.py

```diff
@@ -1355,9 +1356,28 @@ def rewrite_blockwise(inputs):
 
             changed = True
 
+            # Check if this dep was already encountered.
+            # If so, we should append a "counter" label
+            # to the name to avoid fusing distinct indices
+            # into a single dependency within the subgraph
+            # (see: https://github.com/dask/dask/issues/8535)
+            use_dep = dep
+            if dep in seen:
+                try:
+                    _ind = seen[dep][ind]
+                    if _ind:
+                        # We only append a "counting" label for
+                        # `ind` cases after the first ooccurrence
```

Comment:
Were you ale to find a case where both the `dep` and the `ind` were matching and the previous implementation separated them? I was not, unfortunately

Reply:
Hmm - I was thinking `da.add` would capture this case, but apparently it does not. In that case, the simpler `defaultdict(int)` solution may be preferable for now.